jme3-core/src/main/java/com/jme3/scene/Spatial.java

```diff
@@ -811,7 +811,7 @@ public void addControlAt(int index, Control control) {
 
         if (index < numControls) { // re-arrange the list directly
             boolean success = controls.remove(control);
-            assert success;
+            assert success : "Surprising control remove failure. " + control.getClass().getSimpleName() + " from spatial " + getName();
```

Comment:
and `Surprising \"control.remove()\" failure`

Reply:
I would only offer to beware of using string-notation in messages when the value is not actually a string.  In these cases they are both unnecessary but neither would be confusing... but that is not always the case.

Sometimes there are messages like:
Foodebla blah blah bad "This string" deblah blah...

And in that case the quoted string has meaning... because if you instead see:
Foodebla blah blah bad "This string " deblah blah... (note extra space)
Or:
Foodebla blah blah bad ""This string"" deblah blah...

...you have an indication of your error already.

I would argue that none of your examples need strings around them but I could see an argument for wrapping getName() in quotes for all the reasons I suggest the other is confusing.

Reply:
Good point. I'm not married to using quotes; I'm just wondering if there's a good, simple way to differentiate identifiers from the ordinary English words that they resemble, and that are almost - but not quite - grammatically correct in context.

Reply:
No getting away from looking in the code for an exception like this... then it's obvious.

Reply:
Okay, I think you convinced me. Thanks for the discussion.